frontend-ssr/components/tableOfContents/TableOfContents.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import styles from './TableOfContents.module.scss';
+
+interface State {
+    activeTitleChapter: string,
+}
+
+interface Props {
+    chapters: Chapters[],
+}
+
+interface Chapters {
+    title: string,
+    id: string,
+}
+
+class TableOfContents extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeTitleChapter: null,
+    };
+  }
+
+    handleScrollItem = (id) => {
+      let itemToScroll = document.getElementById(id);
+      if (itemToScroll !== null) {
+        itemToScroll.scrollIntoView(
+          {
+            behavior: 'smooth',
+          },
+        );
+      } else {
+        console.warn('TableOfContents: expected to find element with id="id" to scroll to');
+      }
+      this.setState({
+        activeTitleChapter: id,
+      });
+    }
+
+    render() {
+      const { chapters } = this.props;
+      const { activeTitleChapter } = this.state;
+      return (
+        <ul className={styles.tableOfContents}>
+          {chapters.map((item) => (
+            <li
+              key={item.id}
+              className={activeTitleChapter === item.id ? styles.active : ''}
+            >
+              <button
```

Comment:
As mentioned above, change this to an `<a>`. But we can keep the classes which we need.